modules/optimizing-the-vllm-runtime.adoc

```diff
@@ -0,0 +1,169 @@
+. To configure the vLLM model-serving runtime for multi-modal inferencing, add the following arguments as shown:
++
+[source]
+----
+containers:
+  - args:
+      - --trust-remote-code
+----
++
+[NOTE]
+====
+Ensure that you use the `--trust-remote-code` argument with models from trusted sources. 
+====
+. Click Update.
++
+The *Serving runtimes* page opens and shows the list of runtimes that are installed. Observe that the custom model-serving runtime you updated is shown.
+. For speculative decoding, you must additionally redeploy the `InferenceService` custom resource definition (CRD) for the vLLM model-serving runtime as follows:
+.. Log in to the OpenShift CLI.
+.. List the available inference services in your namespace.
++
+[source]
+----
+oc get -n <namespace> isvc
+----
+Note the name of the `InferenceService` that needs to be re-deployed.
+.. Save the `InferenceService` manifest to a YAML file:
++
+[source]
+----
+oc get -n <namespace> isvc <inference-service-name> -o yaml > inferenceservice.yml
+----
+Replace the placeholder values with your own.
+.. Delete the exisiting `InferenceService`:
++
+[source]
+----
+oc delete -n <namespace> isvc <inference-service-name>
+----
+Replace the placeholder values with your own.
+.. Deploy the modified `InferenceService` CRD using the YAML file that you saved.
++
+[source]
+----
+oc apply -f inferenceservice.yml
+----
+.. Optional: Check the status of the `InferenceService` deployment as follows:
++
+[source]
+----
+oc get pod,isvc -n <namespace>
+----
+Replace placeholder values with your own.
+ifdef::upstream[]
+. Deploy the model by using the custom runtime  as described in {odhdocshome}{default-url}/serving_models/serving-large-models_serving-large-models#adding-a-custom-model-serving-runtime-for-the-single-model-serving-platform_serving-large-models[Deploying models on the single-model serving platform].
+endif::[]
+ifndef::upstream[]
+. Deploy the model by using the custom runtime  as described in {rhoaidocshome}{default-url}/serving_models/serving-large-models_serving-large-models#adding-a-custom-model-serving-runtime-for-the-single-model-serving-platform_serving-large-models[Deploying models on the single-model serving platform].
+endif::[]
+
+.Verification
+
+* If you have configured the vLLM model-serving runtime for speculative decoding, use the following example command to verify API requests to your deployed model:
++
+[source]
+----
+curl -v https://<inference_endpoint_url>:443/v1/chat/completions 
+-H "Content-Type: application/json" 
+-H "Authorization: Bearer <token>"
+----
+* If you have configured the vLLM model-serving runtime for multi-modal inferencing, use the following example command to verify API requests to the Vision-Language Model (VLM) that you have deployed:
++
+[source]
+----
+curl -v https://<inference_endpoint_url>:443/v1/chat/completions 
```

Comment:
Could also add an example with the `/v1/completions` endpoint. In that case

```
-d '{"model":"<model_name>", "prompt": "<text>"}' ...
```

Reply:
@dtrifiro AFAIK `:443/v1/completions` does not handle image url or anything in such, So based on that i don't think we can add completion endpoints

Reply:
Well it's there and part of the OpenAI API spec, although it's being deprecated: https://platform.openai.com/docs/guides/completions

Since it should be equivalent to the chat API (abeit a bit simpler), I guess we can leave it out